### PR TITLE
Adding readme commands to build with the Go Docker container

### PR DESCRIPTION
**Pull request by surfingdoggo**
Building with Docker is incredibly useful, so here are the commands for anyone wishing to do so.

Resolves https://github.com/opentofu/opentofu/issues/562

## Target Release

1.6.0-dev

### NEW FEATURES | UPGRADE NOTES | ENHANCEMENTS | BUG FIXES | EXPERIMENTS

 

-

### CODE REVIEW

**Comment by surfingdoggo** [comment]
https://github.com/opentofu/opentofu/issues/562 closes

**Comment by surfingdoggo** [comment]
closes https://github.com/opentofu/opentofu/issues/562

**Comment by chriscunningham-cts** [comment]
Given that there is a devcontainer now as of #371, the docs should mention that as well.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,6 +84,16 @@ go test ./internal/command/...
 go test ./internal/addrs
 ```
 
+### Build with Docker
+
+Install [Docker](https://docs.docker.com/engine/install/) and, if you're on Linux, run the [post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/).
+
+```
+docker run --rm -v "$PWD":/usr/src/opentofu -w /usr/src/opentofu golang:1.20.7 go build -v -buildvcs=false .
```
Comment (cube2222) [inline CONTRIBUTING.md:92]: If this is not on linux, then I believe you'll also want to pass GOOS and GOARCH explicitly. I think it's worth adding this here. So
```
docker run --rm -v "$PWD":/usr/src/opentofu -w /usr/src/opentofu golang:1.20.7 GOOS=linux GOARCH=amd64 go build -v -buildvcs=false .
```

**Review by cube2222 — COMMENTED** [review]
Thanks for the contribution @surfingdoggo!

One minor comment left.

**Comment by chriscunningham-cts** [comment]
It's worth noting that if you use the included devcontainer then you can build directly with `go build`. I'll see if I can put together a PR that fully documents that process.

**Comment by thumperward** [comment]
> It's worth noting that if you use the included devcontainer then you can build directly with `go build`. I'll see if I can put together a PR that fully documents that process.

See Lost-Newbs/opentofu#1

**Comment by cube2222** [comment]
Closing due to inactivity.

**Pull request closed without merging**